Sources/SwiftParser/Declarations.swift

```diff
@@ -1697,12 +1697,91 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+
+    let fixityText = [Keyword.prefix, .infix, .postfix].map({ $0.defaultText })
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena)
+
+    var fixity = self.missingToken(.prefix)
+    var leftModifiers: [RawDeclModifierSyntax?] = []
+
+    if let modifiers = attrs.modifiers?.elements {
+      for modifier in modifiers {
+        if fixity.presence == .missing,
+          fixityText.contains(modifier.name.tokenText)
+        {
+          fixity = modifier.name
+          break
+        }
+        leftModifiers.append(modifier)
+      }
+    }
+
+    var (unexpectedBeforeOperatorKeyword, operatorKeyword) = self.expect(.keyword(.operator))
+
+    // Sometimes we could not parse the fixity into attr.modifiers
+    // so we need extract it from unexpectedBeforeOperatorKeyword manually
+    if fixity.presence == .missing,
+      let elements = unexpectedBeforeOperatorKeyword?.elements
+    {
```

Comment:
Yeah, that's another way I thought before. But there are some corner cases in the way when I tried to recover to `DeclarationModifier` with `declKeyword` precedence and I could not find a proper workaround with them.  

1. 
```swift
enum foo {
  case a
  static let b
}
```

The modifier `static` here has  `declKeyword` precedence and `case` has `stmtKeyword` precedence, which makes the whole `case a` as an unexpected node.

2. If we promote the `case` to `declKeyword` precedence, codes like
```swift
switch {
  @case
}
```
will generate some broken diagnostic message since `case` has a higher precedence than `}`

Do you have any idea about this?

Reply:
OK, I’ve got one more idea: What you want to do is to raise the recovery precedence of `case` in this case. So, `case` needs to be part of the `TokenSpecSet` that you pass to `canRecoveryTo`. Since `case` is not a declaration modifier though, the way I would go about doing this, is to define a union set of `DeclarationStart` and `DeclarationModifier` just like we do for `ExpressionStart`. You should then be able to check if you can recover to this union set and if you recover to a case in `DeclarationStart`, you just `break` the loop.

Reply:
Cool! This works perfectly! 🤩 Already submitted this change.